### PR TITLE
Added gRPC telemetry instrumentation example.

### PR DESCRIPTION
**Pull request by marvin-hansen**
Resolves issue: https://github.com/autometrics-dev/autometrics-rs/issues/162

Added end-to-end example to instrument gRPC service with graceful shutdown and health service for K8s deployment.
 
Updated example Readme.

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ cargo run --package example-{name of example}
 - [custom-metrics](./custom-metrics/) - Define your own custom metrics alongside the ones generated by autometrics (using any of the metrics collection crates)
 - [exemplars-tracing](./exemplars-tracing/) - Use fields from `tracing::Span`s as Prometheus exemplars
 - [opentelemetry-push](./opentelemetry-push/) - Push metrics to an OpenTelemetry Collector via the OTLP HTTP or gRPC protocol using the Autometrics provided interface
+- [grpc-http](./grpc-http/) - Instrument Rust gRPC services with observability using Tonic, warp, and Autometrics.
```
Comment (mellowagain) [inline examples/README.md:17]:
```suggestion
- [grpc-http](./grpc-http/) - Instrument Rust gRPC services with metrics using Tonic, warp, and Autometrics.
```

File: examples/grpc-http/README.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+Sample output:
+
+```
+73014 ttys002    0:00.25 /Users/.../autometrics-rs/target/debug/grpc-http
+```
+
+In this example, the service runs on PID 73014. Let's send a sigterm signal to shutdown the service. On you system, a different PID will be returned so please use that one instead. 
+
+```bash
+Kill 73014
```
Comment (mellowagain) [inline examples/grpc-http/README.md:87]:
```suggestion
kill 73014
```

File: examples/grpc-http/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    println!("gRPC shutdown complete");
+}
+
+fn print_start(web_addr: &SocketAddr, grpc_addr: &SocketAddr) {
+    println!();
+    println!("Started gRPC server on port {:?}", grpc_addr.port());
+    println!("Started metrics on port {:?}", web_addr.port());
+    println!("Explore autometrics at http://127.0.0.1:6789");
+    println!();
+}
```
Comment (mellowagain) [inline examples/grpc-http/src/main.rs:97]: can u add a newline at the end of the file

File: examples/grpc-http/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+[dependencies]
+autometrics = { path = "../../autometrics", features = ["prometheus-exporter"] }
+prost = "0.12"
+tokio = { version = "1", features = ["full"] }
+tonic = "0.10"
+tonic-health = "0.10"
+warp = "0.3"
+
+[build-dependencies]
+tonic-build = "0.10"
```
Comment (mellowagain) [inline examples/grpc-http/Cargo.toml:16]: can u add a newline at the end of this file as well

**Review by mellowagain — COMMENTED** [review]
hey, thanks for your contribution! a few nits here and there but else it looks good! could you also run `cargo fmt` so the formatting is all good? ill then run the CI on it and merge it if it passes

**Comment by marvin-hansen** [comment]
Sure,

I suspect the JetBrains default code formatter ran over all files instead
of rust fmt since I didn’t updated the project settings.

Let fix those small things once I’m back in the office and I issue an
update probably over the weekend.

Thanks for your review and suggestions.

M

File: examples/grpc-http/src/server.rs [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+// https://docs.autometrics.dev/rust/quickstart
+// Attach the SLO to each API function.
+//
+// Notice, all API functions are instrumented with the same SLO.
+// If you want to have different SLOs for different API functions,
+// You have to create a separate SLO for each API function and instrument
+// each API function individually instead of using the macro on trait level.
+// Docs https://docs.autometrics.dev/rust/adding-alerts-and-slos
+#[tonic::async_trait]
+#[autometrics(objective = API_SLO)]
```
Comment (mellowagain) [inline examples/grpc-http/src/server.rs:43]: can u swap those two around

```suggestion
#[autometrics(objective = API_SLO)]
#[tonic::async_trait]
```

**Review by mellowagain — APPROVED** [review]
last small nit but else looks good

**Pull request merged**